xarray/core/variable.py

```diff
@@ -1765,12 +1766,12 @@ def concat(cls, variables, dim='concat_dim', positions=None,
 
         return cls(first_var.dims, data, attrs)
 
-    def copy(self, deep=True):
+    def copy(self, deep=True, data=None):
         """Returns a copy of this object.
 
-        `deep` is ignored since data is stored in the form of pandas.Index,
-        which is already immutable. Dimensions, attributes and encodings are
-        always copied.
+        `deep` and `data` are ignored since data is stored in the form of
```

Comment:
I'm not sure it makes sense to skip the `data` argument for an IndexVariable.

`deep=True` is safe to ignore, because there's no need to worry about deeply copying data for immutable objects (the pandas.Index).

But that doesn't exclude making a new object with new data.

Reply:
Yeah that is definitely right. I wasn't thinking about it properly. I made the change.